### PR TITLE
Add SM 3.2 to restore compatibility matrix

### LINKED ISSUE #3617: docs: Issue in page Compatibility Matrix
I would like to report an issue in page https://manager.docs.scylladb.com/branch-3.2/restore/compatibility-matrix

### Problem
Shouldn't Scylla Manager 3.2 be listed on this page?

###  Suggest a fix

### PR DESCRIPTION
**Pull request by Michal-Leszczynski**
This change in documentation is already present on master and should be backported to branch 3.2.

Fixes #3617

### CODE REVIEW

File: docs/source/restore/compatibility-matrix.rst [hunk 1]
```diff
@@ -11,6 +11,9 @@ The following table shows which version of Scylla Manager restore task supports
    * - ScyllaDB Manager Version
      - ScyllaDB Open Source Version
      - ScyllaDB Enterprise Version
+   * - 3.2
+     - 5.0, 5.1, 5.2
```
Comment (tzach) [inline docs/source/restore/compatibility-matrix.rst:15]: Can we smoke test and add 5.4 here?
RC1 is available

Comment (Michal-Leszczynski) [reply docs/source/restore/compatibility-matrix.rst:15]: I will trigger the following tests for SM 3.2 and Scylla 5.4.0-rc1:
- [SM repo tests](https://github.com/scylladb/scylla-manager/actions/runs/6763490314/)
- [Dtests](https://jenkins.scylladb.com/view/scylla-manager/job/manager-master/job/manager-dtest/1346/)
- [centos-sanity-test](https://jenkins.scylladb.com/view/scylla-manager/job/manager-master/job/centos-sanity-test/) - queued

Comment (Michal-Leszczynski) [reply docs/source/restore/compatibility-matrix.rst:15]: Tests are failing, but it looks like a problem with config changes. I will investigate that.
E.g. with some changes to testing `scylla.yaml` in SM repo, SM tests pass when cluster is created successfully, but the process of building it sometimes fails.

**Comment by Michal-Leszczynski** [comment]
Restore matrix with SM 3.2 is already present on 3.2 branch.

**Pull request closed without merging**